pyeffects/Either.py

```diff
@@ -85,7 +85,16 @@ def __init__(self, value: A) -> None:
 
     def left(self):
         return self.value
-
+    
+    def __eq__(self, __value: object) -> bool:
+        isLeft = isinstance(__value, self.__class__)
```

Comment:
you can check this using the `is_right()` and `is_left()` functions.

The implementation could be something as simple as:

```
return self.is_left() == other.is_left && self.is_right() == self.is_right && self.value == other.value
```